servers/rest-services/src/main/java/org/projectnessie/services/rest/RestV2TreeResource.java

```diff
@@ -0,0 +1,238 @@
+  }
+
+  @Override
+  public SingleReferenceResponse createReference(
+      String name, Reference.ReferenceType type, Reference reference)
+      throws NessieNotFoundException, NessieConflictException {
+    String fromRefName = null;
+    String fromHash = null;
+    if (reference != null) {
+      fromRefName = reference.getName();
+      fromHash = reference.getHash();
+    }
+
+    Reference created = tree().createReference(name, type, fromHash, fromRefName);
+    return SingleReferenceResponse.builder().reference(created).build();
+  }
+
+  @Override
+  public SingleReferenceResponse getReferenceByName(GetReferenceParams params)
+      throws NessieNotFoundException {
+    return SingleReferenceResponse.builder()
+        .reference(tree().getReferenceByName(params.getRefName(), params.fetchOption()))
+        .build();
+  }
+
+  @Override
+  public EntriesResponse getEntries(String ref, EntriesParams params)
+      throws NessieNotFoundException {
+    Reference reference = Reference.fromPathString(ref, Reference.ReferenceType.BRANCH);
+    return tree().getEntries(reference.getName(), reference.getHash(), null, params.filter());
+  }
+
+  @Override
+  public LogResponse getCommitLog(String ref, CommitLogParams params)
+      throws NessieNotFoundException {
+    Reference reference = Reference.fromPathString(ref, Reference.ReferenceType.BRANCH);
+    return tree()
+        .getCommitLog(
+            reference.getName(),
+            params.fetchOption(),
+            params.startHash(),
+            reference.getHash(),
+            params.filter(),
+            params.maxRecords(),
+            params.pageToken());
+  }
+
+  @Override
+  public DiffResponse getDiff(DiffParams params) throws NessieNotFoundException {
+    Reference from = Reference.fromPathString(params.getFromRef(), Reference.ReferenceType.BRANCH);
+    Reference to = Reference.fromPathString(params.getToRef(), Reference.ReferenceType.BRANCH);
+    return diff().getDiff(from.getName(), from.getHash(), to.getName(), to.getHash());
+  }
+
+  @Override
+  public SingleReferenceResponse assignReference(
+      Reference.ReferenceType type, String ref, Reference assignTo)
+      throws NessieNotFoundException, NessieConflictException {
+    Reference reference = Reference.fromPathString(ref, type);
+    tree().assignReference(type, reference.getName(), reference.getHash(), assignTo);
+    return SingleReferenceResponse.builder().reference(reference).build();
+  }
+
+  @Override
+  public SingleReferenceResponse deleteReference(Reference.ReferenceType type, String ref)
+      throws NessieConflictException, NessieNotFoundException {
+    Reference reference = Reference.fromPathString(ref, type);
+    tree().deleteReference(type, reference.getName(), reference.getHash());
+    return SingleReferenceResponse.builder().reference(reference).build();
+  }
+
+  @Override
+  public ContentResponse getContent(ContentKey key, String ref) throws NessieNotFoundException {
```

Comment:
Looks like this method has no test coverage

Reply:
added `AbstractV2ResteasyTest` to cover this